roles/cluster_compare/tasks/upstream.yml

```
@@ -0,0 +1,102 @@
+---
+- name: Get the latest tools release tag
+  ansible.builtin.uri:
+    url: "https://api.github.com/repos/openshift/kube-compare/releases/latest"
+    method: GET
+    return_content: true
+    status_code: 200
+  register: _cc_github_response
+  no_log: "{{ cc_logging }}"
+
+- name: Download and extract binary files
+  vars:
+    cc_version: "{{ _cc_github_response.json.tag_name | default('v0.6.0', true) }}"
+    cc_base_repo: https://github.com/openshift/kube-compare/releases/download
+    cc_bins:
+      - kube-compare_addon_tools_linux_amd64.tar.gz
+      - kube-compare_linux_amd64.tar.gz
+  ansible.builtin.unarchive:
+    src: "{{ cc_base_repo }}/{{ cc_version }}/{{ binary }}"
+    dest: "{{ cc_work_dir }}"
+    remote_src: true
+    mode: '0755'
+  register: _cc_extract
+  retries: 1
+  delay: 10
+  until: _cc_extract is not failed
+  loop_control:
+    loop_var: binary
+  loop: "{{ cc_bins }}"
+  no_log: "{{ cc_logging }}"
+
+- name: Get cluster version
+  community.kubernetes.k8s_info:
+    api: config.openshift.io/v1
+    kind: ClusterVersion
+    name: version
+  register: _cc_cluster_version
+  no_log: "{{ cc_logging }}"
+
+- name: Set cluster version
+  vars:
+    current_ver_query: "history[?state=='Completed'] | [0].version"
+    full_ver: "{{ _cc_cluster_version.resources[0].status | json_query(current_ver_query) }}"
+    current_ver: "{{ full_ver | regex_search('^\\d+\\.\\d+') }}"
```

The regex does not seem to be right, what about:
```suggestion
    current_ver: "{{ full_ver | regex_search('^\d+\.\d+') }}"
```

---

Actually the new is not working, it complains about not been escaped properly. regex_search('^\\d+\\.\\d+') works fine

---

I see, is because yaml interpreting `\` before the actual filter. :+1: